services/api/blockassembler.go

```diff
@@ -0,0 +1,83 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/attestantio/go-eth2-client/spec/capella"
+	"github.com/flashbots/go-utils/cli"
+	"github.com/flashbots/go-utils/jsonrpc"
+	"github.com/flashbots/mev-boost-relay/common"
+)
+
+var assemblyRequestTimeout = time.Duration(cli.GetEnvInt("BLOCKASSEMBLY_TIMEOUT_MS", 60000)) * time.Millisecond
+
+type IBlockAssembler interface {
+	Send(context context.Context, request *common.BlockAssemblerRequest) (*capella.ExecutionPayload, error, error)
+}
+
+type BlockAssembler struct {
+	cv               *sync.Cond
+	counter          int64
+	blockAssemblyURL string
+	client           http.Client
+}
+
+func NewBlockAssembler(blockAssemblyURL string) *BlockAssembler {
+	return &BlockAssembler{
+		cv:               sync.NewCond(&sync.Mutex{}),
+		blockAssemblyURL: blockAssemblyURL,
+		client: http.Client{ //nolint:exhaustruct
+			Timeout: assemblyRequestTimeout,
+		},
+	}
+}
+
+func (b *BlockAssembler) Send(context context.Context, request *common.BlockAssemblerRequest) (payload *capella.ExecutionPayload, requestErr, validationErr error) {
```

Comment:
any way we can refactor this to use the same send as the block sim client? it looks very similar.

Reply:
true, we can do that! I will again do these things in one huge clean up PR! first priority is to get everything to work!